lib/TransportRPCClient.js

```diff
@@ -165,14 +168,37 @@ class TransportRPCClient extends Base.TransportRPCClient {
     const _opts = {
       headers: {
         _gr: this.format([req.rid, req.key]),
+        connection: 'close',
+        'transfer-encoding': 'chunked',
         ...addHeaders
       },
       timeout: req.opts.timeout
     }
 
     const opts = this.getOpts(_opts, this.conf.secure)
-    const stream = this.socket.post(opts)
-    return stream
+
+    // node-fetch does not support multiplex communication, no response is received until body is fully streamed
+    const reqStream = new PassThrough()
+    const resStream = new PassThrough()
+
+    const url = new URL(opts.url)
+    const httpReq = (this.conf.secure ? https : http).request({
+      ...opts,
+      hostname: url.hostname,
+      port: url.port,
+      protocol: url.protocol,
+      path: url.pathname,
+      method: 'POST'
+    }, (res) => {
+      res.pipe(resStream)
```

Comment:
maybe we could use pipeline/pump to properly destroy things if error occurs? `pipe` doesn't do that afaik